ChampionTej05/GoWebapp
# Add automated testing to the web app

Description:

This pull request adds automated testing to the Go web application. Specifically, it introduces a set of unit tests and integration tests that cover the main functionality of the application.

The unit tests use the standard Go testing framework to test individual functions and methods of the codebase, ensuring that each component behaves as expected. The integration tests, on the other hand, use the testify library and simulate HTTP requests to test the entire application end-to-end, from the router to the database.

The test suite includes tests for the handlers, the models, and the database layer, covering both happy paths and edge cases. The tests are designed to run in isolation and can be executed with the `go test` command.

This PR also includes some minor refactorings to make the code more testable, such as reducing global state and isolating dependencies.

Overall, this addition of automated testing should improve the quality and reliability of the web app, making it easier to maintain and evolve over time.

```code
ok  	github.com/ChampionTej05/GoWebapp/controllers	0.271s	coverage: 43.1% of statements
ok  	github.com/ChampionTej05/GoWebapp/models	0.332s	coverage: 100.0% of statements
```

**How to Test**
To test this PR, you can follow these steps:

- Clone the repository and checkout the Webapp-Testing branch.
- Install the required dependencies by running go mod download.
- Run the tests by executing go test -v ./....
- Run the application by executing go run main.go.
- Open your web browser and go to http://localhost:8080 to see the application in action.
